### PR TITLE
Prepare object_store 0.5 release

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

re https://github.com/apache/arrow-rs/issues/2620
# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: object_store/CHANGELOG.md [hunk 1]
```diff
@@ -19,52 +19,44 @@
 
 # Changelog
 
-## [object_store_0.4.0](https://github.com/apache/arrow-rs/tree/object_store_0.4.0) (2022-08-10)
+## [object_store_0.5.0](https://github.com/apache/arrow-rs/tree/object_store_0.5.0) (2022-09-08)
 
-[Full Changelog](https://github.com/apache/arrow-rs/compare/object_store_0.3.0...object_store_0.4.0)
+[Full Changelog](https://github.com/apache/arrow-rs/compare/object_store_0.4.0...object_store_0.5.0)
+
+**Breaking changes:**
```
Comment (tustvold) [inline object_store/CHANGELOG.md:26]: These technically aren't breaking changes, but I think they are major enough to justify

File: object_store/.github_changelog_generator [hunk 2]
```diff
@@ -23,5 +23,5 @@ add-sections={"documentation":{"prefix":"**Documentation updates:**","labels":["
 # so that the component is shown associated with the issue
 issue-line-labels=object-store
 # skip non object_store issues
-exclude-labels=development-process,invalid,arrow,parquet,arrow-flight
+exclude-labels=development-process,invalid,arrow,parquet,arrow-flight,parquet-derive
```
Comment (alamb) [inline object_store/.github_changelog_generator:26]: 👍

File: object_store/CHANGELOG.md [hunk 3]
```diff
@@ -19,52 +19,44 @@
...
-- object\_store: Add example in the main documentation readme [\#2202](https://github.com/apache/arrow-rs/issues/2202) [[object-store](https://github.com/apache/arrow-rs/labels/object-store)]
+- IMDSv1 Fallback for S3 [\#2609](https://github.com/apache/arrow-rs/issues/2609) [[object-store](https://github.com/apache/arrow-rs/labels/object-store)]
+- Print Response Body On Error [\#2572](https://github.com/apache/arrow-rs/issues/2572) [[object-store](https://github.com/apache/arrow-rs/labels/object-store)]
+- Coalesce Ranges Parallel Fetch [\#2562](https://github.com/apache/arrow-rs/issues/2562) [[object-store](https://github.com/apache/arrow-rs/labels/object-store)]
+- Support Coalescing Out-of-Order Ranges [\#2561](https://github.com/apache/arrow-rs/issues/2561) [[object-store](https://github.com/apache/arrow-rs/labels/object-store)]
+- object\_store: use different AWS environment variable name in integration [\#2550](https://github.com/apache/arrow-rs/issues/2550) [[object-store](https://github.com/apache/arrow-rs/labels/object-store)]
```
Comment (alamb) [inline object_store/CHANGELOG.md:37]: a minor thing, but I would recommend tagging this one as "development process" as users likely won't care about changes we made to the integration tests

**Review by alamb — APPROVED** [review]
Looks great -- thanks @tustvold ❤️ 

Do you have any idea when you would like to propose a release https://github.com/apache/arrow-rs/issues/2620 still lists "TBD"

I am happy to run the scripts needed to do so

File: object_store/CHANGELOG.md (hunk 3 above)
Comment (tustvold) [reply object_store/CHANGELOG.md:37]: Darn, thought I got them all :sweat_smile:

**Comment by alamb** [comment]
Once this PR is merged, I am happy to cut a release

**Pull request merged**